src/actions/SupplierCreditActions.js

```diff
@@ -0,0 +1,92 @@
+/**
+ * mSupply Mobile
+ * Sustainable Solutions (NZ) Ltd. 2020
+ */
+
+import { batch } from 'react-redux';
+import { createRecord, UIDatabase } from '../database';
+import { refreshData } from '../pages/dataTableUtilities/actions/tableActions';
+import { ROUTES } from '../navigation/constants';
+
+export const SUPPLIER_CREDIT_ACTIONS = {
+  CREATE_FROM_ITEM: 'SupplierCredit/createFromItem',
+  CLOSE: 'SupplierCredit/close',
+  SORT: 'SupplierCredit/sort',
+  EDIT_RETURN_AMOUNT: 'SupplierCredit/editReturnAmount',
+};
+
+const sort = sortKey => ({ type: SUPPLIER_CREDIT_ACTIONS.SORT, payload: { sortKey } });
+
+const close = () => ({ type: SUPPLIER_CREDIT_ACTIONS.CLOSE });
+
+const create = () => (dispatch, getState) => {
+  const { supplierCredit, user } = getState();
+  const { batches } = supplierCredit;
+  const { currentUser } = user;
+
+  // Only work with the batches whose return amount is greater than 0.
+  const batchesToReturn = batches.reduce(
+    (acc, itemBatch) => (itemBatch.returnAmount ? [...acc, itemBatch] : acc),
+    []
+  );
+
+  // Group the batches with a return amount by supplier to make a credit for each
+  // supplier grouping { supplierId1: [batch1, batch2, ... batchn], supplierId2: [...], ...}
+  const batchesGroupedBySupplier = batchesToReturn.reduce((groupings, itemBatch) => {
+    const { id: suppliersId } = itemBatch?.itemBatch.supplier;
```

Comment:
Is this a typo with `itemBatch?.itemBatch`?

Reply:
nah it's technically an object shown in the reducer that is mapped to `{ returnAmount, itemBatch, ... }` - couldnt and can't think of a different name 🤣